float64 roughness
float64 hardness
---
float64[] map
